build-system-abstraction.rst

```diff
@@ -0,0 +1,456 @@
+
+PYTHONPATH
+    Used to control sys.path per the normal Python mechanisms.
+
+Subcommands
+-----------
+
+There are a number of separate subcommands that build systems must support.
+The examples below use a build_command of ``flit`` for illustrative purposes.
+
+build_requires
+    Query build requirements. Build requirements are returned as a UTF-8
+    encoded JSON document with one key ``build_requires`` consisting of a list
+    of dependency specifications [#dependencyspec]_. Additional keys must be
+    ignored. The build_requires command is the only command run without
+    setting up a build environment.
+
+    Example command::
+
+        flit build_requires
+
+metadata
+    Query project metadata.  The metadata and only the metadata should
+    be output on stdout in UTF-8 encoding. pip would run metadata just once to
+    determine what other packages need to be downloaded and installed. The
+    metadata is output as a wheel METADATA file per PEP-427 [#pep427]_.
+
+    Note that the metadata generated by the metadata command, and the metadata
+    present in a generated wheel must be identical.
+
+    Example command::
+
+        flit metadata
+
+wheel -d OUTPUT_DIR
+    Command to run to build a wheel of the project. OUTPUT_DIR will point to
+    an existing directory where the wheel should be output. Stdout and stderr
+    have no semantic meaning. Only one file should be output - if more are
+    output then pip would pick an arbitrary one to consume.
+
+    Example command::
+
+        flit wheel -d /tmp/pip-build_1234
+
+develop [--prefix PREFIX] [--root ROOT]
+    Command to do an in-place 'development' installation of the project.
+    Stdout and stderr have no semantic meaning.
+
+    Not all build systems will be able to perform develop installs. If a build
+    system cannot do develop installs, then it should error when run. Note
+    that doing so will cause use operations like ``pip install -e foo`` to
+    fail.
+
+    The prefix option is used for defining an alternative prefix within the
+    installation root.
+
+    The root option is used to define an alternative root within which the
+    command should operate.
+
+    For instance::
+
+        flit develop --root /tmp/ --prefix /usr/local
+
+    Should install scripts within `/tmp/usr/local/bin`, even if the Python
+    environment in use reports that the sys.prefix is `/usr/` which would lead
+    to using `/tmp/usr/bin/`. Similar logic applies for package files etc.
+
+The build environment
+---------------------
+
+Except for the build_requires command, all commands are run within a build
+environment. No specific implementation is required, but a build environment
+must achieve the following requirements.
+
+1. All dependencies specified by the project's build_requires must be
+   available for import from within ``$PYTHON``.
+
+1. All command-line scripts provided by the build-required packages must be
+   present in ``$PATH``.
```

Comment:
Should ``$PATH`` be mentioned in the environment variables that are provided up above?